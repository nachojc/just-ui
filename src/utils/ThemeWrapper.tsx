import { ReactNode } from "react";
import { ThemeProvider } from "styled-components";
import theme from "../styles/themes";

type ThemeName = keyof typeof theme;

type options = {
  themeName: ThemeName;
};
export const wrapWithTheme = (children: ReactNode, options?: options) => {
  const opt: options = { ...{ themeName: "defaultTheme" }, ...options };
  return <ThemeProvider theme={theme[opt.themeName]}>{children}</ThemeProvider>;
};
