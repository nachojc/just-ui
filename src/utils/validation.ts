export const hasText = (value?: string): boolean => {
  return typeof value === "string" && value?.trim().length > 0;
};
