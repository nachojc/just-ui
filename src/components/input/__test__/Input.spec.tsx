import { render } from "@testing-library/react";
import { Input } from "..";
import React from "react";
import { wrapWithTheme } from "../../../utils/ThemeWrapper";

describe("Button", () => {
  it("render", async () => {
    const btn = render(wrapWithTheme(<Input />));
    expect(btn).toBeTruthy();
  });
});
