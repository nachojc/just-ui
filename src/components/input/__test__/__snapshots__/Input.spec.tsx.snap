// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input default render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF leasZd"
    >
      <input
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input small 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF leasZd"
    >
      <div
        class="sc-fAUdSK kWQtov"
      >
        £
      </div>
      <input
        class="sc-gtLWhw hLxVsq"
      />
      <div
        class="sc-dntaoT eOVwam"
      >
        .00
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with Label 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    >
      Input
    </label>
    <div
      class="sc-egkSDF leasZd"
    >
      <input
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with disabled 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF bUWDbo"
      disabled=""
    >
      <div
        class="sc-fAUdSK fhttrK"
        disabled=""
      >
        £
      </div>
      <input
        class="sc-gtLWhw hHAYCT"
        disabled=""
      />
      <div
        class="sc-dntaoT XDsPf"
        disabled=""
      >
        .00
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with display asBlock 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    />
    <div
      class="sc-egkSDF leasZd"
    >
      <input
        class="sc-gtLWhw jpkqNe"
      />
    </div>
    <div
      class="sc-ivxoEo eixUTS"
    />
  </div>
</DocumentFragment>
`;

exports[`Input with error 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
      error="no text"
    >
      Input
    </label>
    <div
      class="sc-egkSDF bklFls"
      error="no text"
    >
      <input
        class="sc-gtLWhw jpkqNe"
        error="no text"
      />
    </div>
    <div
      class="sc-ivxoEo bSyMPD"
      error="no text"
    >
      no text
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with left icon 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    >
      Input
    </label>
    <div
      class="sc-egkSDF leasZd"
    >
      <div
        class="sc-fAUdSK cfPCdj"
      >
        <svg
          class="left"
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
          />
        </svg>
      </div>
      <input
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with left text 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    >
      Input
    </label>
    <div
      class="sc-egkSDF leasZd"
    >
      <div
        class="sc-fAUdSK cfPCdj"
      >
        £
      </div>
      <input
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with right icon 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    >
      Input
    </label>
    <div
      class="sc-egkSDF leasZd"
    >
      <input
        class="sc-gtLWhw jpkqNe"
      />
      <div
        class="sc-dntaoT hEWJjW"
      >
        <svg
          class="right"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M7 20h4c0 1.1-.9 2-2 2s-2-.9-2-2m-2-1h8v-2H5zm11.5-9.5c0 3.82-2.66 5.86-3.77 6.5H5.27c-1.11-.64-3.77-2.68-3.77-6.5C1.5 5.36 4.86 2 9 2s7.5 3.36 7.5 7.5m-2 0C14.5 6.47 12.03 4 9 4a5.51 5.51 0 0 0-5.5 5.5c0 2.47 1.49 3.89 2.35 4.5h6.3c.86-.61 2.35-2.03 2.35-4.5m6.87-2.13L20 8l1.37.63L22 10l.63-1.37L24 8l-1.37-.63L22 6zM19 6l.94-2.06L22 3l-2.06-.94L19 0l-.94 2.06L16 3l2.06.94z"
              fill="currentColor"
              fill-rule="nonzero"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input with right text 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <label
      class="sc-ghWlax kvLWXm"
      data-testid="input-label"
    >
      Input
    </label>
    <div
      class="sc-egkSDF leasZd"
    >
      <input
        class="sc-gtLWhw jpkqNe"
      />
      <div
        class="sc-dntaoT hEWJjW"
      >
        .00
      </div>
    </div>
  </div>
</DocumentFragment>
`;
